packages/astro/src/actions/index.ts

```diff
@@ -3,12 +3,24 @@ import type { Plugin as VitePlugin } from 'vite';
 import type { AstroIntegration } from '../@types/astro.js';
 import { viteID } from '../core/util.js';
 import { ACTIONS_TYPES_FILE, RESOLVED_VIRTUAL_MODULE_ID, VIRTUAL_MODULE_ID } from './consts.js';
+import { AstroUserError } from '../core/errors/errors.js';
+
+const serverOutputs = ['server', 'hybrid'];
 
 export default function astroActions(): AstroIntegration {
 	return {
 		name: VIRTUAL_MODULE_ID,
 		hooks: {
 			async 'astro:config:setup'(params) {
+				if (!serverOutputs.includes(params.config.output)) {
+					const error = new AstroUserError(
```

Comment:
You could use a normal `AstroError` here. If you fear that the error won't last, then you can use a not defined one, but if you think it will last definitely document it! I think it'll be a fairly common wonder of users why actions need output server.